docs/docs/authentication.md

```diff
@@ -4,98 +4,120 @@ description: Set up an authentication provider
+The `useAuth` hook provides a streamlined interface to your auth provider's client SDK.
+Much of what the functions it returns does is self explanatory, but the options they take depend on the auth provider:
 
-```jsx
+| Name              | Description                                                                                                                                                                 |
+| :---------------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `client`          | The client instance used in creating the auth provider                                                                                                                      |
+| `currentUser`     | An object containing information about the current user as set on the `api` side, or if the user isn't authenticated, `null`                                                |
+| `getToken`        | Returns a JWT                                                                                                                                                               |
+| `hasRole`         | Determines if the current user is assigned a role like `"admin"` or assigned to any of the roles in a list                                                                  |
+| `isAuthenticated` | A boolean indicating whether or not the user is authenticated                                                                                                               |
+| `loading`         | If auth is loading                                                                                                                                                          |
+| `logIn`           | Logs a user in                                                                                                                                                              |
+| `logOut`          | Logs a user out                                                                                                                                                             |
+| `reauthenticate`  | Refetch auth data and context                                                                                                                                               |
+| `signUp`          | Signs a user up                                                                                                                                                             |
+| `userMetadata`    | An object containing the user's metadata (or profile information) fetched directly from an instance of the auth provider client, or if the user isn't authenticated, `null` |
+
+### Protecting routes
+
+You can require that a user be authenticated to navigate to a route by wrapping it in the `Private` component.
```

Comment:
Maybe also mention `<Set private` here. I think that will make the example further down that uses Set less confusing :)

Reply:
See https://github.com/redwoodjs/redwood/pull/7229/commits/1df28b6263060100e4544f1ded52ea2fc8a9ac33